vscode/src/commands.ts

```diff
@@ -167,200 +170,228 @@ const commandsMap: (state: ExtensionState) => {
           return;
         }
 
-        const kaiAgent = new KaiInteractiveWorkflow();
-        const agentInit = kaiAgent.init({
+        // Initialize workflow if agent mode is enabled
+        // Create array to store all diffs
+        const allDiffs: { original: string; modified: string; diff: string }[] = [];
+
+        // Set the state to indicate we're fetching a solution
+
+        await state.workflowManager.init({
           model: model,
           workspaceDir: state.data.workspaceRoot,
-          fsCache: state.kaiFsCache,
           solutionServerClient: state.solutionServerClient,
         });
 
-        // revert the changes back to on-disk
-        // state.kaiFsCache.on("cacheInvalidated", async (path) => {
-        //   // TODO (pgaikwad) - revert the changes
-        //   // think about edge cases like if the document is already open by the user etc
-        // });
+        // Get the workflow instance
+        workflow = state.workflowManager.getWorkflow();
+        // Track processed message tokens to prevent duplicates
+        const processedTokens = new Set<string>();
 
         // TODO (pgaikwad) - revisit this
         // this is a number I am setting for demo purposes
         // until we have a full UI support. we will only
         // process child issues until the depth of 1
         const maxTaskManagerIterations = 1;
```

Comment:
is the maxIterations being used anymore?

Reply:
Yeah it is passed  into ProcessMessage.ts where i moved the task logic.